### PR TITLE
Audit Intro & What is libp2p

### PR DESCRIPTION
**Pull request by DannyS03**
## Context

- Getting started style doc for libp2p and combined with intro
- Moves index material on the intro doc to their own sections
- Concepts -> Core components

Part of <--to open auditing issue-->

## Latest preview

Please view the latest Fleek preview [here](https://bafybeicam5kau2f56qvlwzwif4xzitkpp7y4ls3kcytvbqrypjsy2ipdra.on.fleek.co/introduction/)

### CODE REVIEW

File: content/introduction/_index.md [hunk 1]
```diff
@@ -1,60 +1,187 @@
...
+that enable the development of peer-to-peer network applications. 
 
-## Tutorials
+**The networking protocol of web3**
 
-If you want to dive in, check out our collection of [tutorials](/tutorials/), which will help guide you through your explorations of libp2p.
+Because of libp2p's peer-to-peer and distributed architecture, most of the 
+needs and considerations that the current web was built on no longer apply.
+Moving from a location-addressed system to a peer-to-peer, content-addressed 
+system presents many challenges. The internet, such as it is, with firewalls
```
Comment (marten-seemann) [inline content/introduction/_index.md:24]: libp2p doesn't really care about location-addressing vs content-addressing.

File: content/introduction/_index.md [hunk 2]
```diff
@@ -1,60 +1,187 @@
...
 
-## Tutorials
+**The networking protocol of web3**
 
-If you want to dive in, check out our collection of [tutorials](/tutorials/), which will help guide you through your explorations of libp2p.
+Because of libp2p's peer-to-peer and distributed architecture, most of the 
+needs and considerations that the current web was built on no longer apply.
+Moving from a location-addressed system to a peer-to-peer, content-addressed 
+system presents many challenges. The internet, such as it is, with firewalls 
+and NATs, was designed to [securely] provide data by relying on trust assumptions,
```
Comment (marten-seemann) [inline content/introduction/_index.md:25]: incomplete sentence

File: content/introduction/_index.md [hunk 3]
```diff
@@ -1,60 +1,187 @@
...
+**The networking protocol of web3**
 
-If you want to dive in, check out our collection of [tutorials](/tutorials/), which will help guide you through your explorations of libp2p.
+Because of libp2p's peer-to-peer and distributed architecture, most of the 
+needs and considerations that the current web was built on no longer apply.
+Moving from a location-addressed system to a peer-to-peer, content-addressed 
+system presents many challenges. The internet, such as it is, with firewalls 
+and NATs, was designed to [securely] provide data by relying on trust assumptions, 
+Clients must trust the service providers they connect to and the data they provide, 
+despite servers needing to verify their identity using certificate authorities.
```
Comment (marten-seemann) [inline content/introduction/_index.md:27]: This seems like a motivation for IPFS, not libp2p.

File: content/introduction/_index.md [hunk 4]
```diff
@@ -1,60 +1,187 @@
...
-While libp2p has several implementations, it is fundamentally a set of protocols for peer identity, discover, routing, transport and more.
+### What problems can libp2p solve?
 
-See the [specifications section](/reference/specs/) for details.
+While peer-to-peer networks have many advantages over the client-server model, 
+there are unique challenges that require careful thought and practice to overcome.
 
-### Implementations
+libp2p lets all users preserve their network identity, overcome network censorship, 
+and communicate over different transfer protocols.
```
Comment (marten-seemann) [inline content/introduction/_index.md:54]:
```suggestion
and communicate over different transport protocols.
```

File: content/introduction/_index.md [hunk 5]
```diff
@@ -1,60 +1,187 @@
...
+### Data transmission
 
-Get in touch with other members of the libp2p community who are building tools and applications with libp2p! You can ask questions, discuss new ideas, or get support for problems at [libp2p discussion forums](https://discuss.libp2p.io/), but you can also [hop on IRC](/community/irc/) for a quick chat.
+The transport layer is at the foundation of libp2p, which is responsible for 
+the actual transmission and receipt of data from one peer to another. There are many 
+ways to send data across networks in use today, with more in development and still more yet 
+to be designed. 
 
-See the other links in the community section for more information about meetings, events, apps people are building, and more.
+libp2p provides a simple [interface](https://github.com/libp2p/js-libp2p-interfaces)
```
Comment (marten-seemann) [inline content/introduction/_index.md:76]: This interface only applies to js-libp2p. We shouldn't mention it here.

File: content/introduction/_index.md [hunk 6]
```diff
@@ -1,60 +1,187 @@
...
+2. The `PeerId` allows anyone to retrieve the public key for the identified 
+   peer, which enables secure communication between peers.
 
-libp2p is an open-source community project. While [Protocol Labs](https://protocol.ai) is able to sponsor some of the work around it, much of the design, code, and effort is contributed by volunteers and community members like you. If you’re interested in helping improve libp2p, check the [contributing](/contributing/) guide to get started.
+### Secure communication
 
-If you are diving in to contribute new code, make sure you check both the [contribution guidelines](https://github.com/libp2p/community/blob/master/CONTRIBUTE.md) and the style guide for your language ([Go](https://github.com/ipfs/community/blob/master/CONTRIBUTING_GO.md), [JavaScript](https://github.com/ipfs/community/blob/master/CONTRIBUTING_JS.md)).
+We must be able to send and receive data between peers securely, 
+where we can trust the [identity](#peer-identity) of the peer we're 
+communicating with no third party can read our conversation or alter it in flight.
```
Comment (marten-seemann) [inline content/introduction/_index.md:96]: This sentence is broken.

File: content/introduction/_index.md [hunk 7]
```diff
@@ -1,60 +1,187 @@
...
 
+libp2p supports *upgrading* a connection provided by a [transport protocol](#transport) 
+into a securely encrypted channel. The process is flexible and supports multiple 
+methods of encrypting communication.
```
Comment (marten-seemann) [inline content/introduction/_index.md:100]: Better: All libp2p connections are encrypted and authenticated. Some transport protocols are encrypted at the transport layer (e.g. QUIC), for other protocols libp2p runs a cryptographic handshake on top of an unencrypted connection (e.g. TCP).

File: content/introduction/_index.md (hunk 5 above)
Comment (DannyS03) [reply content/introduction/_index.md:76]: Ah, I've come across some of the deprecated repos (in go for instance) that point to the js interface page

File: content/introduction/_index.md (hunk 7 above)
Comment (DannyS03) [reply content/introduction/_index.md]: Adapted

File: content/introduction/_index.md (hunk 6 above)
Comment (DannyS03) [reply content/introduction/_index.md]: fixed

File: content/introduction/_index.md (hunk 3 above)
Comment (DannyS03) [reply content/introduction/_index.md]: Right, updated.

File: content/introduction/_index.md (hunk 2 above)
Comment (DannyS03) [reply content/introduction/_index.md]: Updated.

File: content/introduction/_index.md (hunk 1 above)
Comment (DannyS03) [reply content/introduction/_index.md]: Updated.

File: content/introduction/_index.md [hunk 8]
```diff
@@ -1,60 +1,187 @@
...
+start.
 
-Head over to [What is libp2p?](/introduction/what-is-libp2p/) for an introduction to the basics of libp2p and an overview of the problems it addresses.
+## What is libp2p?
 
+libp2p is a modular system of *protocols*, *specifications* and *libraries* 
+that enable the development of peer-to-peer network applications. 
 
-## Tutorials
+**The networking protocol of web3**
```
[the remark this answers is not in the document]
Comment (DannyS03) [reply content/introduction/_index.md:19]: I just removed the one-liner as the following paragraph mentions the same thing

**Comment by DannyS03** [comment]
thanks @marten-seemann, addressed majority

File: content/introduction/_index.md (hunk 5 above)
Comment (DannyS03) [reply content/introduction/_index.md]: Generalized to reference libp2p-spec.

File: content/introduction/_index.md [hunk 9]
```diff
@@ -1,60 +1,188 @@
...
```
Comment (marten-seemann) [inline content/introduction/_index.md:19]:
```suggestion
**A p2p networking stack**
```

libp2p is also used outside of the blockchain space, and I'd like our branding to reflect that we're (aspiring to be a) general-purpose p2p networking stack.

File: content/introduction/_index.md [hunk 10]
```diff
@@ -1,60 +1,188 @@
...
+This is an ever-growing space, so don't be surprised if things change over time. 
+If you notice something missing or have other ideas for improving this documentation, 
+please [reach out to let us know][help_improve_docs].
 
-## Community
+### Data transmission
 
-Get in touch with other members of the libp2p community who are building tools and applications with libp2p! You can ask questions, discuss new ideas, or get support for problems at [libp2p discussion forums](https://discuss.libp2p.io/), but you can also [hop on IRC](/community/irc/) for a quick chat.
+The transport layer is at the foundation of libp2p, which is responsible for 
+the actual transmission and receipt of data from one peer to another. There are many
```
Comment (marten-seemann) [inline content/introduction/_index.md:72]:
```suggestion
transmitting and receiving bytes between two peers. There are many 
```

File: content/introduction/_index.md [hunk 11]
```diff
@@ -1,60 +1,188 @@
...
+please [reach out to let us know][help_improve_docs].
 
-## Community
+### Data transmission
 
-Get in touch with other members of the libp2p community who are building tools and applications with libp2p! You can ask questions, discuss new ideas, or get support for problems at [libp2p discussion forums](https://discuss.libp2p.io/), but you can also [hop on IRC](/community/irc/) for a quick chat.
+The transport layer is at the foundation of libp2p, which is responsible for 
+the actual transmission and receipt of data from one peer to another. There are many 
+ways to send data across networks in use today, with more in development and still more yet 
+to be designed.
```
Comment (marten-seemann) [inline content/introduction/_index.md:74]: Should we mention existing transports (TCP, QUIC, WebSocket, WebTransport, WebRTC)? Might help to make things more concrete.

**Review by marten-seemann — COMMENTED** [review]
Thanks for making the changes @DannyS03! Just a few small things, then this should be good to go.

File: content/introduction/_index.md (hunk 11 above)
Comment (DannyS03) [reply content/introduction/_index.md:75]: That's good, added for more context.

**Review by marten-seemann — APPROVED** [review]
Looks good! Thank you @DannyS03!

**Pull request merged**